### DISCUSSION TITLE
Ikea Vindstyrka Air Quality Monitor No Precision

### DISCUSSION THREAD

**Discussion by jamesonuk** [Koenkk/zigbee2mqtt#18764] [Devices]
Not sure if this is a limitation of the device or a configuration issue but I have just got a Vindstyrka Air Quality Monitor (https://www.zigbee2mqtt.io/devices/E2112.html#ikea-e2112) and this is only reporting temperature and humidity as whole numbers.

I have set the precision to 2/3 digits in device specific setting but it makes no difference.

I have seen the front end issue where by clicking in the calibration box ends up with value is not a number exceptions but I have cleaned these up in the yaml configuration and restarted Z2M to rule out anything around that.

The device config appears pretty much standard so I am guessing this is just that the device only reports whole numbers?

**Comment by SaschaRH** [comment]
I can confirm this behavior:
{
    "humidity": 56,
    "linkquality": 255,
    "pm25": 2,
    "temperature": 26,
    "update": {
        "installed_version": 16777233,
        "latest_version": 16777233,
        "state": "idle"
    },
    "update_available": false,
    "voc_index": 112
}

I have set the precision to 1 digit. Homekit shows xx,0. :-D

**Reply by kkossev** [reply]
It is the device itself that reports the temperature and the humidity as whole numbers.

**Comment by shrx** [comment]
Duplicate of https://github.com/Koenkk/zigbee2mqtt/issues/17636
